redhat-access/app/models/redhat_access/concerns/host_managed_extensions.rb

```diff
@@ -12,7 +12,7 @@ module ClassMethods
         def search_by_plan_id(key, operator, value)
           insights_plan_runner = ForemanAnsible::InsightsPlanRunner.new(Organization.current, value.to_i)
           hostname_rules_relation = insights_plan_runner.hostname_rules(insights_plan_runner.playbook)
-          host_ids = Host::Managed.where(:name => hostname_rules_relation.keys).pluck(:id)
+          host_ids = Host::Managed.where(:name => hostname_rules_relation.keys.map{|h| h.downcase}).pluck(:id)
```

Comment:
looks reasonable, you can write `.map(&:downcase)` instead of `.map{|h| h.downcase}` to keep it more readable

Reply:
Thank you @ares  for the hint. After troubleshooting we found that `insights-client` utility considers $HOSTNAME (which is upper case always), when registering to cloud.redhat.com, the same is reflected in the insights inventory.  

Here, I am trying to fix the lookup to match the host records from foreman (which are registered in lowercase). However, this is just a part of the problem and by fixing this we faced another problem as explained below.

Automated remediation playbook generated (by selecting `Run Playbook` option) through this approach always creates an empty file (due to hostname mismatch from cloud.redhat.com and what we are trying to lookup from foreman) but downloading the same manually using a `Download playbook` generates the correct playbook. (Note sure how)

I am unsure which approach should we go from here.

Reply:
Download playbook fetches the playbook directly, Foreman reparses it due to modifying the target for each host. Here is where we find Foreman hosts by name https://github.com/theforeman/foreman_ansible/blob/master/app/services/foreman_ansible/insights_plan_runner.rb#L18

We coud probably turn it to case insensitive. Instead of `Host::Managed.find_by(:name => hostname)` we could do `Host::Managed.where('lower(name) = ?', hostname.downcase).first `. @snagoor mind to test this and send a patch to foreman_ansible?

Reply:
@ares, I have tested this and running a playbook from Insights page doesn't seem to invoke the method `run_playbook` mentioned here https://github.com/theforeman/foreman_ansible/blob/a3da7528fd0dcb8ea67a7d5befc42263fe6764b2/app/services/foreman_ansible/insights_plan_runner.rb#L14.

Reply:
@snagoor  this part does not sound right to me - can you clarify ? :
"After troubleshooting we found that insights-client utility considers $HOSTNAME (which is upper case always), when registering to cloud.redhat.com, the same is reflected in the insights inventory."